syntax_and_semantics/structs.md

```diff
@@ -30,4 +30,36 @@ A struct can also include modules and can be generic, just like a class.
 
 A struct is mostly used for performance reasons to avoid lots of small memory allocations when passing small copies might be more efficient.
 
-So how do you choose between a struct and a class? The rule of thumb is that if no instance variable is ever reassigned, i.e. your type is immutable, you could use a struct, otherwise use a class.
+## Example of object passing difference
+
+```crystal
+class Klass
+  property array = ["foo"]
+end
+
+struct Strukt
+  property array = ["foo"]
+end
+
+def modify(object)
+  object.array << "bar"
+  object.array = ["new"]
+end
+
+strukt, klass = Strukt.new, Klass.new
+modify strukt
+modify klass
+puts strukt.array #=> ["foo", "bar"]
+puts klass.array  #=> ["new"]
+```
+
+Explanation:
+- `array` is an `Array` class, thus passed by reference
+- the `array` object can be modified - elements inside it can be add- /removed/modified
+- the `array` object can't be replaced, and remains `["foo", "bar"]` - `Strukt` being a struct, and immutable
+- `Klass` is a class, everything is passed by reference - all values can be replaced no matter the scope
+
+## Choosing between `struct` and `class`
+
+- `struct` can be used at first, which is more efficient
```

Comment:
The languages you listed above have already this notion of passing by value/reference, each one has its own quirks.

Using classes everywhere when they aren't needed is also a bad advice, and could hurt in the future instead of the present.

I'll remove this section. I think the goal of docs is to bring informations, how things works, not advices.

Reply:
No, I think this language reference should really include advise on using structs and classes. Otherwise readers would know that they both exist, but have no clue when to use what?

> could hurt in the future instead of the present.

Where does using classes hurt?

Reply:
Hurt performance on certain cases. No need to add more advises, a link points now to the existing section https://crystal-lang.org/docs/guides/performance.html#use-structs-when-possible

Reply:
Again, performance should not be the main concern when designing a data model.

"Use structs whenever possible" is only valid from a performance point of view. And performance is surely often important, but not everything.